File: backend/utils/email.js
import nodemailer from 'nodemailer';
import dotenv from 'dotenv';

dotenv.config();

// Create reusable transporter
const transporter = nodemailer.createTransport({
    host: process.env.EMAIL_HOST,
    port: process.env.EMAIL_PORT,
    secure: process.env.EMAIL_PORT === '465', // true for 465, false for other ports
    auth: {
        user: process.env.EMAIL_USER,
        pass: process.env.EMAIL_PASSWORD
    }
});

// Send activation email
export const sendActivationEmail = async (email, firstName, activationToken) => {
    const activationUrl = `${process.env.FRONTEND_URL}/activate/${activationToken}`;
    
    const mailOptions = {
        from: process.env.EMAIL_FROM,
        to: email,
        subject: 'Account Activation - E-commerce Platform',
        html: `
            <h2>Welcome to E-commerce Platform, ${firstName}!</h2>
            <p>Thank you for registering. Please click the link below to activate your account:</p>
            <a href="${activationUrl}" style="display: inline-block; padding: 10px 20px; background-color: #007bff; color: white; text-decoration: none; border-radius: 5px;">Activate Account</a>
            <p>This link will expire in 24 hours.</p>
            <p>If you didn't create an account, please ignore this email.</p>
        `
    };

    try {
        await transporter.sendMail(mailOptions);
        console.log('Activation email sent successfully');
    } catch (error) {
        console.error('Error sending activation email:', error);
        throw new Error('Failed to send activation email');
    }
};

// Send password reset email
export const sendPasswordResetEmail = async (email, firstName, resetToken) => {
    const resetUrl = `${process.env.FRONTEND_URL}/reset-password/${resetToken}`;
    
    const mailOptions = {
        from: process.env.EMAIL_FROM,
        to: email,
        subject: 'Password Reset - E-commerce Platform',
        html: `
            <h2>Password Reset Request</h2>
            <p>Hi ${firstName},</p>
            <p>You requested to reset your password. Click the link below to reset it:</p>
            <a href="${resetUrl}" style="display: inline-block; padding: 10px 20px; background-color: #007bff; color: white; text-decoration: none; border-radius: 5px;">Reset Password</a>
            <p>This link will expire in 1 hour.</p>
            <p>If you didn't request a password reset, please ignore this email.</p>
        `
    };

    try {
        await transporter.sendMail(mailOptions);
        console.log('Password reset email sent successfully');
    } catch (error) {
        console.error('Error sending password reset email:', error);
        throw new Error('Failed to send password reset email');
    }
};

// Send order confirmation email
export const sendOrderConfirmationEmail = async (email, firstName, order) => {
    const mailOptions = {
        from: process.env.EMAIL_FROM,
        to: email,
        subject: 'Order Confirmation - E-commerce Platform',
        html: `
            <h2>Order Confirmation</h2>
            <p>Hi ${firstName},</p>
            <p>Thank you for your order! Your order has been received and is being processed.</p>
            <p><strong>Order ID:</strong> ${order._id}</p>
            <p><strong>Total Amount:</strong> $${order.totalPrice.toFixed(2)}</p>
            <p><strong>Status:</strong> ${order.status}</p>
            <p>You can track your order status in your account.</p>
        `
    };

    try {
        await transporter.sendMail(mailOptions);
        console.log('Order confirmation email sent successfully');
    } catch (error) {
        console.error('Error sending order confirmation email:', error);
        // Don't throw error here as order is already created
    }
};

// Send order status update email
export const sendOrderStatusEmail = async (email, firstName, order) => {
    const mailOptions = {
        from: process.env.EMAIL_FROM,
        to: email,
        subject: 'Order Status Update - E-commerce Platform',
        html: `
            <h2>Order Status Update</h2>
            <p>Hi ${firstName},</p>
            <p>Your order status has been updated.</p>
            <p><strong>Order ID:</strong> ${order._id}</p>
            <p><strong>New Status:</strong> ${order.status}</p>
            <p>You can track your order in your account.</p>
        `
    };

    try {
        await transporter.sendMail(mailOptions);
        console.log('Order status email sent successfully');
    } catch (error) {
        console.error('Error sending order status email:', error);
        // Don't throw error
    }
};

File: django_backend/users/utils/email.py
from django.core.mail import send_mail
from django.conf import settings
from django.template.loader import render_to_string
from django.utils.html import strip_tags


def send_activation_email(user, token):
    """Send account activation email"""
    activation_url = f"{settings.FRONTEND_URL}/activate/{token}"
    
    subject = 'Activate Your E-commerce Account'
    html_message = f"""
    <html>
        <body>
            <h2>Welcome to our E-commerce Platform!</h2>
            <p>Hi {user.first_name},</p>
            <p>Thank you for registering. Please click the link below to activate your account:</p>
            <p><a href="{activation_url}">{activation_url}</a></p>
            <p>This link will expire in 24 hours.</p>
            <p>If you didn't create an account, please ignore this email.</p>
            <br>
            <p>Best regards,<br>E-commerce Team</p>
        </body>
    </html>
    """
    plain_message = strip_tags(html_message)
    
    send_mail(
        subject,
        plain_message,
        settings.DEFAULT_FROM_EMAIL,
        [user.email],
        html_message=html_message,
        fail_silently=False,
    )


def send_password_reset_email(user, token):
    """Send password reset email"""
    reset_url = f"{settings.FRONTEND_URL}/reset-password/{token}"
    
    subject = 'Password Reset Request'
    html_message = f"""
    <html>
        <body>
            <h2>Password Reset</h2>
            <p>Hi {user.first_name},</p>
            <p>We received a request to reset your password. Click the link below to reset it:</p>
            <p><a href="{reset_url}">{reset_url}</a></p>
            <p>This link will expire in 1 hour.</p>
            <p>If you didn't request a password reset, please ignore this email.</p>
            <br>
            <p>Best regards,<br>E-commerce Team</p>
        </body>
    </html>
    """
    plain_message = strip_tags(html_message)
    
    send_mail(
        subject,
        plain_message,
        settings.DEFAULT_FROM_EMAIL,
        [user.email],
        html_message=html_message,
        fail_silently=False,
    )


def send_order_confirmation_email(order):
    """Send order confirmation email"""
    subject = f'Order Confirmation - {order.order_number}'
    html_message = f"""
    <html>
        <body>
            <h2>Order Confirmation</h2>
            <p>Hi {order.user.first_name},</p>
            <p>Thank you for your order! Your order has been received and is being processed.</p>
            <p><strong>Order Number:</strong> {order.order_number}</p>
            <p><strong>Total Amount:</strong> ${order.total_amount}</p>
            <p><strong>Shipping Address:</strong><br>
            {order.shipping_full_name}<br>
            {order.shipping_address}<br>
            {order.shipping_city}, {order.shipping_country}</p>
            <p>You can track your order status in your account.</p>
            <br>
            <p>Best regards,<br>E-commerce Team</p>
        </body>
    </html>
    """
    plain_message = strip_tags(html_message)
    
    send_mail(
        subject,
        plain_message,
        settings.DEFAULT_FROM_EMAIL,
        [order.user.email],
        html_message=html_message,
        fail_silently=False,
    )


def send_order_status_email(order):
    """Send order status update email"""
    subject = f'Order Status Update - {order.order_number}'
    html_message = f"""
    <html>
        <body>
            <h2>Order Status Update</h2>
            <p>Hi {order.user.first_name},</p>
            <p>Your order status has been updated.</p>
            <p><strong>Order Number:</strong> {order.order_number}</p>
            <p><strong>Current Status:</strong> {order.get_status_display()}</p>
            <p>You can view your order details in your account.</p>
            <br>
            <p>Best regards,<br>E-commerce Team</p>
        </body>
    </html>
    """
    plain_message = strip_tags(html_message)
    
    send_mail(
        subject,
        plain_message,
        settings.DEFAULT_FROM_EMAIL,
        [order.user.email],
        html_message=html_message,
        fail_silently=False,
    )
